### PR TITLE
[stake][swarm] Add e2e rewards test

### PR DESCRIPTION
**Pull request by igor-aptos**
### Description

Added rewards test:
- we have 4 nodes.
- one leaves validator set
- one is unreliable

We fetch their stakes, and confirm increases match their proposal success rate.

### Test Plan
added test

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3066)

### CODE REVIEW

**Comment by igor-aptos** [comment]
Test issue is resolved, all works now. Cause was that validators are not lexicographically sorted, and AnalyzeValidators was depending on that, fixed it.

**Review by zekun000 — APPROVED** [review]

File: testsuite/smoke-test/src/aptos_cli/validator.rs [hunk 1]
```diff
@@ -69,6 +75,346 @@ async fn test_show_validator_set() {
...
+
+    tokio::time::sleep(Duration::from_secs(5)).await;
+
+    reconfig(
+        &rest_clients[0],
+        &transaction_factory,
+        swarm.chain_info().root_account(),
+    )
+    .await;
+    println!("#0.5");
```
Comment (movekevin) [inline testsuite/smoke-test/src/aptos_cli/validator.rs:185]: These println!s look like they're for debugging. Can we remove them before merging the PR?

**Review by movekevin — APPROVED** [review]

File: testsuite/smoke-test/src/aptos_cli/validator.rs (hunk 1 above)
Comment (igor-aptos) [reply testsuite/smoke-test/src/aptos_cli/validator.rs:185]: :( forgot about those

Comment (igor-aptos) [reply testsuite/smoke-test/src/aptos_cli/validator.rs:185]: But I'll remove the number ones. The rest are useful for root-causing if test breaks.

**Pull request merged**